lms/djangoapps/grades/tests/test_models.py

```diff
@@ -0,0 +1,193 @@
+
+
+class GradesModelTestCase(TestCase):
+    """
+    Base class for common setup of grades model tests.
+    """
+    def setUp(self):
+        super(GradesModelTestCase, self).setUp()
+        self.course_key = CourseLocator(
+            org='some_org',
+            course='some_course',
+            run='some_run'
+        )
+        self.locator_a = BlockUsageLocator(
+            course_key=self.course_key,
+            block_type='problem',
+            block_id='block_id_a'
+        )
+        self.locator_b = BlockUsageLocator(
+            course_key=self.course_key,
+            block_type='problem',
+            block_id='block_id_b'
+        )
+        self.record_a = BlockRecord(unicode(self.locator_a), 1, 10)
+        self.record_b = BlockRecord(unicode(self.locator_b), 1, 10)
+
+
+@ddt.ddt
+class BlockRecordTest(GradesModelTestCase):
+    """
+    Test the BlockRecord model.
+    """
+    def setUp(self):
+        super(BlockRecordTest, self).setUp()
+
+    def test_creation(self):
+        """
+        Tests creation of a BlockRecord.
+        """
+        weight = 1
+        max_score = 10
+        record = BlockRecord(
+            self.locator_a,
+            weight,
+            max_score,
+        )
+        self.assertEqual(record.locator, self.locator_a)
+
+    @ddt.data(
+        (0, 0, "0123456789abcdef"),
+        (1, 10, 'totally_a_real_block_key'),
+        ("BlockRecord is", "a dumb data store", "with no validation"),
+    )
+    @ddt.unpack
+    def test_serialization(self, weight, max_score, block_key):
+        """
+        Tests serialization of a BlockRecord using the to_dict() method.
+        """
+        record = BlockRecord(block_key, weight, max_score)
+        expected = OrderedDict([
+            ("locator", block_key),
+            ("weight", weight),
+            ("max_score", max_score),
+        ])
+        self.assertEqual(expected, record._asdict())
+
+
+class VisibleBlocksTest(GradesModelTestCase):
+    """
+    Test the VisibleBlocks model.
+    """
+    def test_creation(self):
+        """
+        Happy path test to ensure basic create functionality works as expected.
+        """
+        vblocks = VisibleBlocks.objects.create_from_blockrecords([self.record_a])
+        expected_json = json.dumps([self.record_a._asdict()], separators=(',', ':'), sort_keys=True)
+        expected_hash = b64encode(sha256(expected_json).digest())
+        self.assertEqual(expected_json, vblocks.blocks_json)
+        self.assertEqual(expected_hash, vblocks.hashed)
+
+    def test_ordering_matters(self):
+        """
+        When creating new vbocks, a different ordering produces a different 
+        record in the database.  Saving blocks with the same order returns
+        the same record
+        """
+        stored_vblocks = VisibleBlocks.objects.create_from_blockrecords([self.record_a, self.record_b])
+        repeat_vblocks = VisibleBlocks.objects.create_from_blockrecords([self.record_a, self.record_b])
```

Comment:
Thanks.  This test needs to change, though, since we're making ordering irrelevant again.